lib/IPicoPlugin.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * Common interface for Pico plugins
+ *
+ * For a list of supported events see {@link DummyPlugin}; you can use
+ * {@link DummyPlugin} as template for new plugins. For a list of deprecated
+ * events see {@link PicoDeprecated}.
+ *
+ * You SHOULD NOT use deprecated events when implementing this interface.
+ * Deprecated events are triggered by the {@link PicoDeprecated} plugin, if
+ * plugins which don't implement this interface are loaded. You can take
+ * advantage from this behaviour if you want to do something only when old
+ * plugins are loaded. Consequently the old events are never triggered when
+ * your plugin is implementing this interface and no old plugins are present.
+ *
+ * If you're developing a new plugin, you MUST implement IPicoPlugin. If
+ * you're the developer of an old plugin, it is STRONGLY RECOMMENDED to use
+ * the events introduced in Pico 1.0 when releasing a new version of your
+ * plugin. If you want to use any of the new events, you MUST implement
+ * IPicoPlugin and update all other events you use.
+ *
+ * @author  Daniel Rudolf
+ * @link    http://picocms.org
+ * @license http://opensource.org/licenses/MIT
+ * @version 1.0
+ */
+interface IPicoPlugin
```

Comment:
Out of my depth with this one, and have no strong preference either way; however, PicoPluginInterface sounds more natural to me than IPicoPlugin. :speak_no_evil: